blobby/blob.py

```diff
@@ -1,248 +0,0 @@
-from typing import BinaryIO, Callable, List, IO, Type, TypeVar, Iterator
-
-import hashlib
-import io
-import logging
-import os
-import tempfile
-import threading
-from contextlib import contextmanager
-from inspect import getmembers, isfunction, getdoc
-
-from mypy_boto3_s3.service_resource import Bucket as S3Bucket
-from botocore.response import StreamingBody
-
-logger = logging.getLogger(__name__)
-
-T = TypeVar('T')
-
-
-class BlobError(Exception):
-    pass
-
-
-def inherit_docstrings(cls: Type[T]) -> Type[T]:
```

Comment:
My understanding is that this is used here to apply the docstrings from base classes (e.g. `io.RawIOBase`) and apply them to methods that have not had a docstring added? 

I like the idea, however I'm wary that this could accidentally lead to misleading information in the docstrings if the behaviour of a re-defined method here differs from that of the base class. One would have to be sure that the new implementation does not differ from that described in the base class docstring :thinking: 

At the very least I think this should be documented here :+1:

Reply:
I think the tradeoff is well worth it in something like a file-like object, where the API is well-known, broad and desirable to adhere to in any case. I'm adding a docstring to `inherit_docstrings` to stress the requirements :+1: